### PR TITLE
Use ocaml/setup-ocaml@v2

### PR DESCRIPTION
**Pull request by smorimoto**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,6 +1,11 @@
 name: main
 
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
+  schedule:
+    # Prime the caches every Monday
+    - cron: 0 1 * * MON
```
Comment (Sudha247) [inline .github/workflows/main.yml:8]: Hope I'm reading this correctly: runs the CI every Monday at 1 AM. Trying to understand why we'd need to run this cron job on top of the PR and push events.

**Review by Sudha247 — APPROVED** [review]
Thanks @smorimoto! It's useful to update to the latest version of setup-ocaml. I think this is good to merge apart, aside from a clarification on the cron job.

File: .github/workflows/main.yml (hunk 1 above)
Comment (smorimoto) [reply .github/workflows/main.yml:8]: It's an important point. The caching mechanism used in the GitHub Action is that if the cache is not used, the cache will expire in two weeks and you will need a clean build. So this allows you to extend the cache lifetime every week.

Comment (Sudha247) [reply .github/workflows/main.yml:8]: I see, that makes sense. Thanks for the explanation.

**Pull request merged**